cast/src/lib.rs

```diff
@@ -1236,6 +1277,38 @@ impl SimpleCast {
     }
 }
 
+pub struct RawLogs<'a, 'b, M: Middleware> {
+    provider: &'b M,
+    block_stream: FilterWatcher<'a, <M as Middleware>::Provider, TxHash>,
+    filter: Filter,
+}
+impl<'a, 'b, M: Middleware> Unpin for RawLogs<'a, 'b, M> {}
+
+impl<'a, 'b, M: Middleware> Stream for RawLogs<'a, 'b, M>
+where
+    M: Middleware,
+{
+    type Item = Result<Vec<Log>, <M as Middleware>::Error>;
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut TaskContext<'_>,
+    ) -> Poll<Option<Result<Vec<Log>, <M as Middleware>::Error>>> {
+        match Pin::new(&mut self.block_stream).poll_next(cx) {
+            Poll::Ready(block) => {
+                println!("New block! {:?}", block);
+                // Apply the new filter so that it shows logs from the latest block
+                let filter = self.filter.clone().at_block_hash(block.unwrap());
+                // Now that the filter is set, we ask the Provider of any logs based on the filter
+                match Pin::new(&mut self.provider.get_logs(&filter.clone())).poll(cx) {
```

Comment:
this won't work because `self.provider.get_logs(&filter.clone())` creates a new future, right?

so what you're doing here is creating a new future, polling it and if it's pending, dropping it.
you need to store this in your stream as something like `current_get_logs: Option<Pin<Box< dyn Future<Output = ...>>>` that you then can poll

Reply:
@mattsse I am thrilled cause not only was my understanding, but the solution as well. 

It's so exhilarating when you finally "get" it. 

Thanks for all the mentorship :)